### ISSUE TITLE
Video or Screensharing does not work randomly

### ISSUE THREAD

**Issue by mveigel-softkomplett** [nextcloud/spreed#2971]
## Steps to reproduce
1. In my Environment I use a turn Server, but I see the same behavior during direct connections
2. Start Chrome in a Terminal session to see the output
3. Open a Talk video Conversation
4. Sometimes Videos or Screensharing does not work (fuzzy, not reproducable, what works one day might not work in the next conference)
5. When you share a stream, an the other person it just seeing a black screen, you can read the following in the Terminal console:
[11054:3827:0217/170223.325019:ERROR:peer_connection.cc(3810)] AddIceCandidate: ICE candidates can't be added without any remote session description.
[11054:3717:0217/170223.325169:ERROR:rtc_peer_connection_handler.cc(1498)] Error processing ICE candidate.

Usually multiple times.

### Expected behaviour
Video and Screen sharing should work flawlessly

### Actual behaviour
video and/or screensharing doesn't work reliable

## Browser
Chrome 79.0.3945.130 official
Firefox 68.2.0esr

**Microphone available:** yes

**Camera available:** yes

**Operating system:** CentOS 7.7

**Browser name:** Firefox/Chrome/...

**Browser version:** 50.1/55/...

### Browser log

Javascript log:
=========================================================================
Connecting to peer... 
a
talk.js?v=030cb390-0:147 Disconnected. 
a
talk.js?v=030cb390-0:147 PEER CREATED 
a
talk.js?v=030cb390-0:119 OFFER 
Object
to: "lx2w5y07sfczHHmA0Q8xDC/gDM4qvII6BI0UhlORYYeGIFsR3px+6SIZykCYV+NIhiFHHQkj3TS60JCDJT+ZbnNyK0Zb1OPOrIzEHE+ljdcRXxg8wFVcyB14D/nUf1N333dU6Ox7eNa97IOoi6qlGBy2wQEQiWMz/iyUkgLiy4eulB++622IC8EryXF8iI4JFh6p+EZiganF8Y4qb4ue+8FfStztmJTcRYSaj8uobmTNoKtpxm50E+jueqTCfGa"
sid: "1581958275351"
broadcaster: "Trx1yBBysbs23qzuSaglynynkDZSEUKt6D0spTXtUBMOmbWw+Obdv8q37+4tXEuziTaqjaDYy/jUH7YfxT/xuSD65OLZeghoHkJyOSjE1wKALfeL1aAgQ7tHOJUejkIbkmw3hyzKHM1Ay9EPXDg1W14QQqKE9ozMx2kZKpIzy2sG9aCSD3DmAhBZ07tOsdq3PBrPpsTP0/+S5Qk0qaqDEeFy0zfqNoGHOYcT6ZTouVrIupbjO2SNPrBl/LmzFTX"
roomType: "screen"
type: "offer"
payload: {type: "offer", sdp: "v=0
↵o=- 820938707517868775 2 IN IP4 127.0.0.1
↵s=…3786 label:bdf290c4-613b-4f00-a95f-53c0f58a4353
↵", nick: "admin"}
prefix: "webkit"
__proto__: Object
3
nniwi36w:1 Uncaught (in promise) DOMException: Failed to execute 'addIceCandidate' on 'RTCPeerConnection': Error processing ICE candidate
talk.js?v=030cb390-0:993 Leaving call
talk.js?v=030cb390-0:147 Connection closed. 
a
talk.js?v=030cb390-0:147 XXX Remove peer lx2w5y07sfczHHmA0Q8xDC/gDM4qvII6BI0UhlORYYeGIFsR3px+6SIZykCYV+NIhiFHHQkj3TS60JCDJT+ZbnNyK0Zb1OPOrIzEHE+ljdcRXxg8wFVcyB14D/nUf1N333dU6Ox7eNa97IOoi6qlGBy2wQEQiWMz/iyUkgLiy4eulB++622IC8EryXF8iI4JFh6p+EZiganF8Y4qb4ue+8FfStztmJTcRYSaj8uobmTNoKtpxm50E+jueqTCfGa
DevTools failed to parse SourceMap: https://xxxxxxxxxxxxxxxx/apps/spreed/js/talk.js.map

Console Log:
===========================================================================
[19966:6905:0217/175116.923119:ERROR:peer_connection.cc(3810)] AddIceCandidate: ICE candidates can't be added without any remote session description.
[19966:6632:0217/175116.923228:ERROR:rtc_peer_connection_handler.cc(1498)] Error processing ICE candidate.
[19966:6905:0217/175116.923333:ERROR:peer_connection.cc(3810)] AddIceCandidate: ICE candidates can't be added without any remote session description.
[19966:6632:0217/175116.923375:ERROR:rtc_peer_connection_handler.cc(1498)] Error processing ICE candidate.
[19966:6905:0217/175116.923465:ERROR:peer_connection.cc(3810)] AddIceCandidate: ICE candidates can't be added without any remote session description.
[19966:6632:0217/175116.923507:ERROR:rtc_peer_connection_handler.cc(1498)] Error processing ICE candidate.

==========================================================================

<details>
Don't know exactly why this didn't show up for other users, because it happens quite often (Chrome, Firefox). Because it show's up an error message through the console (and in the Javascript Console), I don't think this is aproblem of the setup.
</details>

## Talk app
Talk App: 8.04

**Custom TURN server configured:** yes

**Custom STUN server configured:** yes

## Server configuration

**Operating system**: Ubuntu/RedHat/...

**Web server:** Apache/Nginx

**Database:** MySQL/Maria/SQLite/PostgreSQL

**PHP version:**

**Nextcloud Version:** (see admin page)

**List of activated apps:**

<details>
Enabled:
  - accessibility: 1.4.0
  - activity: 2.11.0
  - admin_audit: 1.8.0
  - bruteforcesettings: 1.5.0
  - cloud_federation_api: 1.1.0
  - comments: 1.8.0
  - contacts: 3.1.8
  - dav: 1.14.0
  - federatedfilesharing: 1.8.0
  - federation: 1.8.0
  - files: 1.13.1
  - files_external: 1.9.0
  - files_pdfviewer: 1.7.0
  - files_rightclick: 0.15.2
  - files_sharing: 1.10.1
  - files_trashbin: 1.8.0
  - files_versions: 1.11.0
  - files_videoplayer: 1.7.0
  - firstrunwizard: 2.7.0
  - logreader: 2.3.0
  - lookup_server_connector: 1.6.0
  - nextcloud_announcements: 1.7.0
  - notifications: 2.6.0
  - oauth2: 1.6.0
  - password_policy: 1.8.0
  - photos: 1.0.0
  - privacy: 1.2.0
  - provisioning_api: 1.8.0
  - recommendations: 0.6.0
  - serverinfo: 1.8.0
  - settings: 1.0.0
  - sharebymail: 1.8.0
  - spreed: 8.0.4
  - support: 1.1.0
  - survey_client: 1.6.0
  - systemtags: 1.8.0
  - text: 2.0.0
  - theming: 1.9.0
  - twofactor_backupcodes: 1.7.0
  - updatenotification: 1.8.0
  - viewer: 1.2.0
  - workflowengine: 2.0.0
Disabled:
  - dashboard
  - encryption
  - ojsxc
  - user_ldap
</details>

**Nextcloud configuration:**

<details>
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": {
            "0": "xxx.yyy.zzz.aaa",
            "2": "nextcloud.softkomplett.de"
        },
        "trusted_proxies": "***REMOVED SENSITIVE VALUE***",
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "overwrite.cli.url": "https:\/\/nextcloud.softkomplett.de",
        "forcessl": true,
        "overwriteprotocol": "https",
        "dbtype": "pgsql",
        "version": "18.0.0.10",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "sendmail",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "memcached.local": "\\OC\\Memcache\\APCu",
        "memcached.distributed": "\\OC\\Memcache\\Memcached",
        "memcached_servers": [
            [
                "localhost",
                11211
            ],
            [
                "xxx.yyy.zzz.aaa",
                11211
            ]
        ],
        "maintenance": false,
        "theme": "",
        "loglevel": 2,
        "mail_smtpsecure": "tls",
        "mail_sendmailmode": "smtp",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "25",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpauth": 1,
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "updater.secret": "***REMOVED SENSITIVE VALUE***",
        "updater.release.channel": "stable",
        "app_install_overwrite": [
            "dashboard",
            "ojsxc"
        ]
    }
}
</details>

### Server log (data/nextcloud.log)
<details>
{"reqId":"XjVT-vndMguKv8tFdLCDBwAAAAc","level":1,"time":"2020-02-01T10:34:03+00:00","remoteAddr":"89.247.33.59","user":"--","app":"updater","method":"GET","url":"/core/ajax/update.php?requesttoken=BIHegbYgDHkrWj%2BtS0O0L%2BUqnrnZ9qoqiwSiyww03R0%3D%3AcLSU6cB4WSEeME2YHCfZYb1hptG3te4c%2FGfUuGBzn34%3D","message":"\\OC\\Updater::updateEnd: Update successful","userAgent":"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.130 Safari/537.36","version":"18.0.0.10"}
{"reqId":"XjVT-vndMguKv8tFdLCDBwAAAAc","level":1,"time":"2020-02-01T10:34:03+00:00","remoteAddr":"89.247.33.59","user":"--","app":"updater","method":"GET","url":"/core/ajax/update.php?requesttoken=BIHegbYgDHkrWj%2BtS0O0L%2BUqnrnZ9qoqiwSiyww03R0%3D%3AcLSU6cB4WSEeME2YHCfZYb1hptG3te4c%2FGfUuGBzn34%3D","message":"\\OC\\Updater::maintenanceDisabled: Turned off maintenance mode","userAgent":"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.130 Safari/537.36","version":"18.0.0.10"}
{"reqId":"XjVT-vndMguKv8tFdLCDBwAAAAc","level":1,"time":"2020-02-01T10:34:03+00:00","remoteAddr":"89.247.33.59","user":"--","app":"updater","method":"GET","url":"/core/ajax/update.php?requesttoken=BIHegbYgDHkrWj%2BtS0O0L%2BUqnrnZ9qoqiwSiyww03R0%3D%3AcLSU6cB4WSEeME2YHCfZYb1hptG3te4c%2FGfUuGBzn34%3D","message":"\\OC\\Updater::resetLogLevel: Reset log level to Warning(2)","userAgent":"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.130 Safari/537.36","version":"18.0.0.10"}
{"reqId":"XjVU@RAPC4jL3FbK0@ngCQAAAAI","level":3,"time":"2020-02-01T10:37:48+00:00","remoteAddr":"89.247.33.59","user":"admin","app":"no app in context","method":"GET","url":"/index.php/settings/apps/update/ojsxc","message":{"Exception":"Symfony\\Component\\Routing\\Exception\\RouteNotFoundException","Message":"Unable to generate a URL for the named route \"ojsxc.javascript.generalConfig\" as such route does not exist.","Code":0,"Trace":[{"file":"/var/www/html/nextcloud/lib/private/Route/Router.php","line":339,"function":"generate","class":"Symfony\\Component\\Routing\\Generator\\UrlGenerator","type":"->","args":["ojsxc.javascript.generalConfig",[],1]},{"file":"/var/www/html/nextcloud/lib/private/URLGenerator.php","line":82,"function":"generate","class":"OC\\Route\\Router","type":"->","args":["ojsxc.javascript.generalConfig",[]]},{"file":"/var/www/html/nextcloud/apps/ojsxc/appinfo/app.php","line":26,"function":"linkToRoute","class":"OC\\URLGenerator","type":"->","args":["ojsxc.javascript.generalConfig"]},{"file":"/var/www/html/nextcloud/lib/private/legacy/app.php","line":266,"args":["/var/www/html/nextcloud/apps/ojsxc/appinfo/app.php"],"function":"require_once"},{"file":"/var/www/html/nextcloud/lib/private/legacy/app.php","line":155,"function":"requireAppFile","class":"OC_App","type":"::","args":["*** sensitive parameter replaced ***"]},{"file":"/var/www/html/nextcloud/lib/private/legacy/app.php","line":966,"function":"loadApp","class":"OC_App","type":"::","args":["*** sensitive parameter replaced ***"]},{"file":"/var/www/html/nextcloud/lib/private/legacy/app.php","line":918,"function":"executeRepairSteps","class":"OC_App","type":"::","args":["*** sensitive parameter replaced ***",{"step":"OCA\\OJSXC\\Migration\\MigrateConfig"}]},{"file":"/var/www/html/nextcloud/lib/private/Installer.php","line":199,"function":"updateApp","class":"OC_App","type":"::","args":["*** sensitive parameters replaced ***"]},{"file":"/var/www/html/nextcloud/apps/settings/lib/Controller/AppSettingsController.php","line":528,"function":"updateAppstoreApp","class":"OC\\Installer","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"/var/www/html/nextcloud/lib/private/AppFramework/Http/Dispatcher.php","line":170,"function":"updateApp","class":"OCA\\Settings\\Controller\\AppSettingsController","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"/var/www/html/nextcloud/lib/private/AppFramework/Http/Dispatcher.php","line":99,"function":"executeController","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[{"__class__":"OCA\\Settings\\Controller\\AppSettingsController"},"updateApp"]},{"file":"/var/www/html/nextcloud/lib/private/AppFramework/App.php","line":125,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[{"__class__":"OCA\\Settings\\Controller\\AppSettingsController"},"updateApp"]},{"file":"/var/www/html/nextcloud/lib/private/AppFramework/Routing/RouteActionHandler.php","line":47,"function":"main","class":"OC\\AppFramework\\App","type":"::","args":["OCA\\Settings\\Controller\\AppSettingsController","updateApp",{"__class__":"OC\\AppFramework\\DependencyInjection\\DIContainer"},{"appId":"*** sensitive parameter replaced ***","_route":"settings.AppSettings.updateApp"}]},{"function":"__invoke","class":"OC\\AppFramework\\Routing\\RouteActionHandler","type":"->","args":[{"appId":"*** sensitive parameter replaced ***","_route":"settings.AppSettings.updateApp"}]},{"file":"/var/www/html/nextcloud/lib/private/Route/Router.php","line":299,"function":"call_user_func","args":[{"__class__":"OC\\AppFramework\\Routing\\RouteActionHandler"},{"appId":"*** sensitive parameter replaced ***","_route":"settings.AppSettings.updateApp"}]},{"file":"/var/www/html/nextcloud/lib/base.php","line":1008,"function":"match","class":"OC\\Route\\Router","type":"->","args":["/settings/apps/update/ojsxc"]},{"file":"/var/www/html/nextcloud/index.php","line":38,"function":"handleRequest","class":"OC","type":"::","args":[]}],"File":"/var/www/html/nextcloud/3rdparty/symfony/routing/Generator/UrlGenerator.php","Line":144,"CustomMessage":"--"},"userAgent":"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.130 Safari/537.36","version":"18.0.0.10"}
</details>

### Turnserver configuration ###
[root@trunk ~]# cat /etc/turnserver/turnserver.conf | grep -v '^#' 
listening-port=80
tls-listening-port=443
listening-ip=xxx.yyy.zzz.bbb
relay-device=eth1
relay-ip=xxx.yyy.zzz.aaa
fingerprint
use-auth-secret
static-auth-secret=**************************************************************
static-auth-secret=*****************************
realm=***************
total-quota=100
stale-nonce
cert=/usr/local/etc/turn_server_cert.pem
pkey=/usr/local/etc/turn_server_pkey.pem
cipher-list="ECDH+AESGCM:DH+AESGCM:ECDH+AES256:DH+AES256:ECDH+AES128:DH+AES:RSA+AES:RSA+3DES:!ADH:!AECDH:!MD5*"
dh-file=/etc/turnserver/dhparam.pem
no-stdout-log
log-file=/var/log/turn.log
simple-log
no-loopback-peers
no-multicast-peers
pidfile="/var/run/turnserver/turnserver.pid"
proc-user=root
proc-group=root
no-cli
cli-password=***********************************
no-tlsv1
no-tlsv1_1

#### Trunserver log ####

0: log file opened: /var/log/turn.log
0: pid file created: /var/run/turnserver/turnserver.pid
0: IO method (main listener thread): epoll (with changelist)
0: WARNING: I cannot support STUN CHANGE_REQUEST functionality because only one IP address is provided
0: Wait for relay ports initialization...
0:   relay xxx.yyy.zzz.bbb initialization...
0:   relay xxx.yyy.zzz.bbb initialization done
0: Relay ports initialization done
0: IO method (general relay thread): epoll (with changelist)
0: turn server id=1 created
0: IO method (general relay thread): epoll (with changelist)
0: turn server id=0 created
0: Total General servers: 2
0: SQLite DB connection success: /var/db/turndb
0: IO method (auth thread): epoll (with changelist)
0: IO method (auth thread): epoll (with changelist)
0: IO method (admin thread): epoll (with changelist)
0: Keep GID: root(0)
0: Keep UID: root(0)

**Comment by mveigel-softkomplett** [comment]
After some more investigation it looks like a completely different issue. During deeper Inspection I found out several things:

- uflag (ind the candidate offer) is NIL on some connections which didn't work
- the remoteDescription is set, the error message seems to be somewhat missleading
- it looks like not all messages about the iceConnections are received (although all are sended by the app - not tested by tcpdump) - so this needs some further investigation
- i'm not sure, but it looks like the the problem might have been caused by the number of Ice candidates. My Test environment has about 6 IP Adresses defined.

Doing some further investigation in the morning.
